### ISSUE TITLE
Addressable dependency is pretty heavy, can it be made optional? 

### ISSUE THREAD

**Issue by SamSaffron** [sdsykes/fastimage#45]
```
retained memory by gem
-----------------------------------
  49751033  activesupport-4.1.8
  10572625  rubygems
   8561818  activerecord-4.1.8
   5367093  2.1.2.discourse/lib
   5178758  actionpack-4.1.8
   4456813  discourse/lib
   3661938  activemodel-4.1.8
   3142081  bundler-1.6.3
   2353753  railties-4.1.8
   1660286  2.1.0
   1564355  actionview-4.1.8
   1424547  mime-types-1.25.1
   1165100  message_bus-1.0.6
   1027115  addressable-2.3.6

```

```
retained objects by gem
-----------------------------------
     79476  activesupport-4.1.8
     57752  rubygems
     35331  actionpack-4.1.8
     25410  mime-types-1.25.1
     22530  2.1.0
     19698  activerecord-4.1.8
     12301  bundler-1.6.3
      9986  addressable-2.3.6
```

Its 1 mb of memory retained + 10k or so objects just to include it. can the work be done simply with the URI standard library?

**Comment by sdsykes** [comment]
See here for discussion: https://github.com/sdsykes/fastimage/pull/46

**Comment by sdsykes** [comment]
Fixed by 31042836d44e
